### ISSUE TITLE
kamailio crash when use http_query

### ISSUE THREAD

**Issue by situnan** [kamailio/kamailio#468]
1.file:modules/utils/functions.c
2.Problem Description：Array Overflow  and  View Code Label：wsong@grandstream.cn 

```
size_t write_function( void *ptr, size_t size, size_t nmemb, void *stream_ptr)
{
	http_res_stream_t *stream = (http_res_stream_t *) stream_ptr;

	stream->buf = (char *) pkg_realloc(stream->buf, stream->curr_size + 
			(size * nmemb) + 1);

	if (stream->buf == NULL) {
		LM_ERR("cannot allocate memory for stream\n");
		return CURLE_WRITE_ERROR;
	}

	memcpy(&stream->buf[stream->pos], (char *) ptr, (size * nmemb));

	stream->curr_size += ((size * nmemb) + 1);
	stream->pos += (size * nmemb);

//begin modify wsong@grandstream.cn 
// reason:stream->buf size begin with 0 
//	stream->buf[stream->pos + 1] = '\0';
stream->buf[stream->pos ] = '\0';
//end modify wsong@grandstream.cn


	return size * nmemb;
}
```

**Comment by que273** [comment]
This issue has been fixed on both master and stable branch 4.3. Please upgrade to the latest version.

**Comment by skrusty** [comment]
Does this also address issues with http_query(): failed to perform curl (23). Out of memory?

```
ERROR: utils [functions.c:57]: write_function(): cannot allocate memory for stream
ERROR: utils [functions.c:150]: http_query(): failed to perform curl (23)
```

Had this happen today, not sure if it's related? Running 4.2
